addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2739,6 +2739,194 @@ options.registry.ReplaceImage = options.Class.extend({
+        }
+        let visibilityId = '';
+        const onlyAttributes = [];
+        const hideAttributes = [];
+        for (const attr of this.$target[0].attributes) {
+            if (attr.name.startsWith('data-visibility-') && !attr.name.startsWith('data-visibility-rule') && !attr.name.endsWith('-id')) {
+                const shortName = attr.name.replace('data-visibility-', '');
+                let dataName, values;
+                if (shortName === 'lang') {
+                    dataName = 'lang';
+                    values = attr.value.replace(/_/g, '-').split(',');
+                } else {
+                    dataName = 'data-' + shortName;
+                    values = attr.value.split(',');
+                }
+                const hideFor = this.$target[0].attributes['data-visibility-rule-' + shortName] && this.$target[0].attributes['data-visibility-rule-' + shortName].value === 'hide';
+                if (hideFor) {
+                    hideAttributes.push({ name: dataName, values: values });
+                } else {
+                    onlyAttributes.push({ name: dataName, values: values });
+                }
+                visibilityId += shortName + this.$target[0].attributes['data-visibility-' + shortName + '-id'].value + (hideFor ? 'h' : '');
+            }
+        }
+        /* Creating CSS selectors based on those attributes, the reducers combine the attributes
+            * Important to note how the CSS Rule works : We hide the content that match the selector. Even for "Only option"
+            * Therefore we hide when the HTML tag does not contain the values we're looking for
+            */
+        let selectors = '';
+        for (const attribute of onlyAttributes) {
+            const selector = attribute.values.reduce((acc, cv) => acc += `:not([${attribute.name}="${cv}"])`, 'html') + ` [data-visibility-id="${visibilityId}"]`;
+            selectors += selector + ', ';
+        }
+        for (const attribute of hideAttributes) {
+            const selector = attribute.values.reduce((acc, cv, i, a) => acc += `html[${attribute.name}="${cv}"] [data-visibility-id="${visibilityId}"]${i !== a.length - 1 ? ',' : ''}`, '');
+            selectors += selector + ', ';
+        }
+        selectors = selectors.slice(0, -2);
+        if (selectors !== '') {
+            this._applyStyleToDom(selectors);
+        }
+
+        this.$target[0].dataset['visibilityId'] = visibilityId;
+        return this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Add a "display: none;" rule with the corresponding selector to a style tag in the #wrap of the page
+     * @param {string} selector a CSS selector
+     */
+
+    _applyStyleToDom: function (selector) {
+        const wrap = document.getElementById('wrap');
```

Comment:
It is not guaranteed there is a wrap in each page (even if it is probably the case) but overal, the wrap element is not necessarily the element you are editing. If the user edits a product description, the content will be saved in that model field... so should the associated style tag. You can probably find the editable element with `this.$target.closest('.o_editable')` ... but maybe to review in a few days when the new editor is merged.